joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/passes/ast/LambdaExpressionTests.scala

```diff
@@ -0,0 +1,402 @@
+              expr.methodFullName shouldBe Operators.conditional
+            case result => fail(s"Expected return conditional, but got $result")
+          }
+          fallBack.name shouldBe "fallback"
+        case result => fail(s"Expected lambda body with single return but got $result")
+      }
+    }
+
+    "create locals for captured identifiers in the lambda method" in {
+      cpg.typeDecl.name("Foo").method.name(".*lambda.*").local.sortBy(_.name) match {
+        case Seq(fallbackLocal: Local) =>
+          fallbackLocal.name shouldBe "fallback"
+          fallbackLocal.code shouldBe "fallback"
+          fallbackLocal.typeFullName shouldBe "string"
+        case result => fail(s"Expected single local for fallback but got $result")
+      }
+    }
+
+    "create closure bindings for captured identifiers" in {
+      cpg.all.collectAll[ClosureBinding].sortBy(_.closureOriginalName) match {
+        case Seq(fallbackClosureBinding) =>
+          val fallbackLocal = cpg.method.name(".*lambda.*").local.name("fallback").head
+          fallbackClosureBinding.closureBindingId shouldBe fallbackLocal.closureBindingId
+
+          fallbackClosureBinding._refOut.l match {
+            case List(capturedParam: MethodParameterIn) =>
+              capturedParam.name shouldBe "fallback"
+              capturedParam.method.fullName shouldBe "Foo.foo:void(string,string)"
+            case result => fail(s"Expected single capturedParam but got $result")
+          }
+
+          fallbackClosureBinding._captureIn.l match {
+            case List(outMethod: MethodRef) =>
+              outMethod.typeFullName shouldBe "Test0.cpp:<global>.Foo.foo.<lambda>0:string(string)"
+              outMethod.methodFullName shouldBe "Test0.cpp:<global>.Foo.foo.<lambda>0:string(string)"
+            case result => fail(s"Expected single METHOD_REF but got $result")
+          }
+        case result => fail(s"Expected 1 closure binding for captured variables but got $result")
+      }
+    }
+
+    "create a typeDecl node inheriting from correct interface" in {
+      cpg.typeDecl.name(".*lambda.*").l match {
+        case List(lambdaDecl) =>
+          lambdaDecl.name shouldBe "<lambda>0"
+          lambdaDecl.fullName shouldBe "Test0.cpp:<global>.Foo.foo.<lambda>0:string(string)"
+          lambdaDecl.inheritsFromTypeFullName should contain theSameElementsAs List("std.function")
+        case result => fail(s"Expected a single typeDecl for the lambda but got $result")
+      }
+    }
+  }
+
+  "lambdas capturing this in method" should {
+    val cpg = code("""
+        |class Foo {
+        |  public:
+        |    int firstDirty;
+        |    void foo() {
+        |      bar(l, [this] { return this->firstDirty == nullptr; });
+        |    }
+        |}
+        |""".stripMargin)
+
+    "ref this correctly" in {
+      val List(thisId) = cpg.identifier.nameExact("this").l
+      val List(refsTo) = thisId.refsTo.collectAll[MethodParameterIn].l
```

Comment:
This ref edge is crossing method boundaries, which is invalid. It should reference a local that captures the outer `this`. (See also the other new comment.)

Reply:
Right, also fixed now.